cog.c

```diff
@@ -289,31 +289,23 @@ platform_setup (CogShell *shell)
 
     g_debug ("%s: Platform name: %s", __func__, s_options.platform_name);
 
-    if (!s_options.platform_name)
-        return FALSE;
-
-    g_autofree char *platform_soname =
-        g_strdup_printf ("libcogplatform-%s.so", s_options.platform_name);
-    g_clear_pointer (&s_options.platform_name, g_free);
-
-    g_debug ("%s: Platform plugin: %s", __func__, platform_soname);
-
-    g_autoptr(CogPlatform) platform = cog_platform_new ();
-    if (!cog_platform_try_load (platform, platform_soname)) {
-        g_warning ("Could not load: %s (possible cause: %s).\n",
-                   platform_soname, strerror (errno));
+    g_autoptr(GError) error = NULL;
+    CogPlatform *platform = cog_platform_new(s_options.platform_name, &error);
+    if (!platform) {
+        g_warning("Cannot create platform: %s).\n", error->message);
```

Comment:
Maybe on purpose, but you don't need a trailing `\n`.

Reply:
Oh, and also there is a stray `)`, let's fix this.